### ISSUE TITLE
fix(html-spec): dl content model should allow zero groups (empty dl is valid)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3592]
## Describe the bug

The `permitted-contents` rule requires at least one `dt`+`dd` group inside `<dl>`, but the HTML spec says "Zero or more groups". An empty `<dl></dl>` is valid.

## Root cause

In `spec.dl.jsonc`, the dt+dd group pattern uses `oneOrMore` but the spec says "zero or more groups", so it should be `zeroOrMore`.

## Code Example

```html
<dl></dl>
```

## The current behavior

```
Require one or more elements. (Need "dt")
```

## The expected behavior

No violations reported.

## Spec reference

> Either: **Zero or more groups** each consisting of one or more dt elements followed by one or more dd elements
> — https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element

## Affected package

`@markuplint/html-spec` — dl element content model

**Comment by YusukeHirao** [comment]
### Investigation result

The root cause is that the content model spec uses `oneOrMore` for dt+dd groups, but the HTML spec says "zero or more groups". Simply changing to `zeroOrMore` in spec.dl.jsonc does not work because the `permitted-contents` engine's `choice` handler doesn't recognize a `zeroOrMore` pattern matching zero times as a valid choice branch.

The issue is in the content model engine (`packages/@markuplint/rules/src/permitted-contents/order.ts`) — when `choice` is evaluated and children are empty, neither branch matches because `choice` requires at least one branch to actively match, and `zeroOrMore` matching zero elements is not treated as an active match by `choice`.

This requires a fix in the `permitted-contents` rule engine, not just the spec data.

**Comment by YusukeHirao** [comment]
v5.0 リリースブロッカー（A群）として #3684 に整理。dl 空コンテナの false positive。
